spec/lib/api/filter_spec.rb

```diff
@@ -406,4 +406,52 @@
       expect(actual.exp).to eq(expected)
     end
   end
+
+  describe "#split_filter_string (private)" do
+    let(:model) { Vm }
+
+    {
+      # all operators, and a few curve balls
+      "name=1"      => [[], "name", "=", "1"],
+      "name=>1"     => [[], "name", "=", ">1"],
+      "name<1"      => [[], "name", "<", "1"],
+      "name<<1"     => [[], "name", "<", "<1"],
+      "name>1"      => [[], "name", ">", "1"],
+      "name>>1"     => [[], "name", ">", ">1"],
+      "name==1"     => [[], "name", "==", "1"],
+      "name!=1"     => [[], "name", "!=", "1"],
+      "name=~1"     => [[], "name", "=~", "1"],
+      "name!~1"     => [[], "name", "!~", "1"],
+      "name!==1"    => [[], "name", "!==", "1"],
+      # funky formatting
+      " name = x "  => [[], "name", "=", "x"],
+      "name=  "     => [[], "name", "=", ""],
+      "name= ' ' "  => [[], "name", "=", "' '"],
+      "host.name=x" => [["host"], "name", "=", "x"],
+    }.each do |str, (associations, column, eoperator, evalue)|
+      it "parses #{str}" do
+        field, operator, value = split_filter_string(str)
+
+        # currently, can not compare one MiqExpression::Target with another
+        expect([operator, value]).to eq([eoperator, evalue])
+        expect([field.associations, field.column]).to eq([associations, column])
+      end
+    end
+
+    [
+      'name^bb',
+      '=bb',
+      'name',
+    ].each do |str|
+      it "complains about '#{str}'" do
+        expect do
+          split_filter_string(str)
+        end.to raise_error(Api::BadRequestError, /Unknown operator specified/)
+      end
+    end
+
+    def split_filter_string(str)
+      described_class.new("", Vm).send(:split_filter_string, str)
```

Comment:
we have `model` defined as a `let` above, but we hardcoded Vm here.  I think we don't need the let anoymore.